homeassistant/components/tuya/__init__.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python3
+"""Support for Tuya Smart devices."""
+
+import itertools
+import json
+import logging
+from typing import Any
+
+from tuya_iot import (
+    ProjectType,
+    TuyaDevice,
+    TuyaDeviceListener,
+    TuyaDeviceManager,
+    TuyaHomeManager,
+    TuyaOpenAPI,
+    TuyaOpenMQ,
+    tuya_logger,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .aes_cbc import AES_ACCOUNT_KEY, KEY_KEY, XOR_KEY, AesCBC as Aes
+from .const import (
+    CONF_ACCESS_ID,
+    CONF_ACCESS_SECRET,
+    CONF_APP_TYPE,
+    CONF_COUNTRY_CODE,
+    CONF_ENDPOINT,
+    CONF_PASSWORD,
+    CONF_PROJECT_TYPE,
+    CONF_USERNAME,
+    DOMAIN,
+    TUYA_DEVICE_MANAGER,
+    TUYA_DISCOVERY_NEW,
+    TUYA_HA_DEVICES,
+    TUYA_HA_TUYA_MAP,
+    TUYA_HOME_MANAGER,
+    TUYA_MQTT_LISTENER,
+    TUYA_SETUP_PLATFORM,
+    TUYA_SUPPORT_HA_TYPE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    vol.All(
+        cv.deprecated(DOMAIN),
+        {
+            DOMAIN: vol.Schema(
+                {
+                    vol.Required(CONF_PROJECT_TYPE): int,
+                    vol.Required(CONF_ENDPOINT): cv.string,
+                    vol.Required(CONF_ACCESS_ID): cv.string,
+                    vol.Required(CONF_ACCESS_SECRET): cv.string,
+                    CONF_USERNAME: cv.string,
+                    CONF_PASSWORD: cv.string,
+                    CONF_COUNTRY_CODE: cv.string,
+                    CONF_APP_TYPE: cv.string,
+                }
+            )
+        },
+    ),
+    extra=vol.ALLOW_EXTRA,
+)
+
+# decrypt or encrypt entry info
+
+
+def entry_decrypt(hass: HomeAssistant, entry: ConfigEntry, init_entry_data):
+    """Decrypt code from config entry."""
+    aes = Aes()
+    # decrypt the new account info
+    if XOR_KEY in init_entry_data:
+        _LOGGER.info("tuya.__init__.exist_xor_cache-->True")
+        key_iv = aes.xor_decrypt(init_entry_data[XOR_KEY], init_entry_data[KEY_KEY])
```

Comment:
It makes no sense to encrypt data in the config entry and then storing the key next to the encrypted data. Please remove the encryption.

Reply:
I think this works if the hacker doesn't know where is the program code.

Reply:
If the hacker can look in this file, knows it's a key for Tuya and knows how to use that key to login it will also know how to decrypt.

Reply:
We should drop this. We don't do this for other integrations. If this is someting desired, we should apply it to all config entries.

Reply:
Like I said in my previous comment, let's drop this.